R/zchunk_L161.nonghg_en_ssp_R_S_T_Y.R

```diff
@@ -0,0 +1,311 @@
+      bind_rows(L114.bcoc_tgej_R_en_S_F_2000 %>%
+                  mutate(year = emissions.GAINS_BASE_YEAR)) %>%
+      # Add GAINS regions and sectors
+      left_join_error_no_match(A_regions %>% select(GCAM_region_ID, GAINS_region), by = "GCAM_region_ID") %>%
+      left_join(GCAM_sector_tech %>% select(supplysector, subsector, stub.technology, IIASA_sector),
+                by = c("supplysector", "subsector", "stub.technology")) %>%
+      # Remove non-IIASA sectors and technologies with 0 emissions factor in base year. No reason to read in future zeroes.
+      filter(!is.na(IIASA_sector), value != 0) %>%
+      rename(base_year = year, base_value = value) %>%
+      left_join(GAINS_emfact_scaler, by = c("GAINS_region", "IIASA_sector", "Non.CO2")) %>%
+      # Scale L111/L114 emissions factors to GAINS scalers
+      mutate(emfact = base_value * scaler) %>%
+      left_join_error_no_match(pcgdp %>% select(GCAM_region_ID, region_grouping), by = "GCAM_region_ID") %>%
+      na.omit() %>%
+      select(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, GAINS_region, IIASA_sector,
+             scenario, year, emfact, region_grouping)
+
+    # Create list of countries with strong regulation based on elec_coal SO2 emissions factor
+    coal_so2 <- tibble(GCAM_region_ID = seq(1, 32)) %>%
+      left_join(
+        emfact_scaled %>%
+          filter(year == emissions.GAINS_YEARS[length(emissions.GAINS_YEARS)]
+                 , IIASA_sector == "elec_coal", Non.CO2 == "SO2", scenario == "CLE"),
+        by = "GCAM_region_ID") %>%
+      mutate(policy = if_else(emfact <= emissions.COAL_SO2_THRESHOLD, "strong_reg", "weak_reg"),
+             policy = replace(policy, region_grouping == "low", "low")) %>%
+      replace_na(list(policy = "low")) %>%
+      select(GCAM_region_ID, policy)
+
+    # Group SSPs by whether we process them the same
+    SSP_groups <- tibble(SSP_group = c("1&5","2","3&4"))
+
+    # Add the rules for each region, gas, technology
+    EF_rules <- emfact_scaled %>%
+      # This is to have same starting point as old code, which begins with NA omitted 2010 values
+      filter(year == emissions.GAINS_YEARS[1]) %>%
+      na.omit %>%
+      select(-year, -emfact, -scenario) %>%
+      distinct() %>%
+      # Repeat for future years and SSP groups
+      repeat_add_columns(tibble(year = c(2010, 2030, 2050, 2100))) %>%
+      repeat_add_columns(SSP_groups) %>%
+      # Join with policy type, but only for SSP group 2
+      left_join_error_no_match(coal_so2, by = "GCAM_region_ID") %>%
+      mutate(policy = replace(policy, SSP_group != "2", NA)) %>%
+      # Join with rules-use left_join b/c there are NA values in A61_emfact_rules
+      left_join(A61_emfact_rules, by = c("region_grouping", "year", "SSP_group", "policy"))
+
+    # Create a tibble with just marker region values
+    # Marker region is Western Europe (13) - some values will be set to its emissions factors in future
+    marker_region <- 13
```

Comment:
actually, @russellhz since we already have a `gcam.USA_CODE` in the constants file, please change this to 

```R
marker_region <- gcam.WESTERN_EUROPE_CODE
```